src/UI/Implementation/Component/Button/Toggle.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+/* Copyright (c) 2018 Thomas Famula <famula@leifos.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Implementation\Component\Button;
+
+use ILIAS\UI\Component as C;
+use ILIAS\UI\Implementation\Component\ComponentHelper;
+use ILIAS\UI\Implementation\Component\JavaScriptBindable;
+use ILIAS\UI\Implementation\Component\Triggerer;
+use ILIAS\UI\Implementation\Component\TriggeredSignal;
+
+class Toggle extends Button implements C\Button\Toggle {
+	use ComponentHelper;
+	use JavaScriptBindable;
+	use Triggerer;
+
+	/**
+	 * @var string
+	 */
+	protected $label;
+
+	/**
+	 * @var bool
+	 */
+	protected $is_on;
+
+	/**
+	 * @var string|null
+	 */
+	protected $action_off;
+
+	protected $action_off_signal;
+	protected $action_on_signal;
+
+	public function __construct($label, $action, $action_off, $is_on)
+	{
+		$this->checkStringArg("label", $label);
+		$this->checkStringOrSignalArg("action", $action);
+		$this->checkStringOrSignalArg("action_off", $action_off);
+		$this->checkBoolArg("is_on", $is_on);
+
+		$this->label = $label;
+
+		if (is_string($action)) {
+			$this->action = $action;
+		}
+		else {
+			$this->action = null;
+			$this->action_on_signal = new TriggeredSignal($action, "click");
+		}
+
+		if (is_string($action_off)) {
+			$this->action_off = $action_off;
+		}
+		else {
+			$this->action_off = null;
+			$this->action_off_signal = new TriggeredSignal($action_off, "click");
```

Comment:
What is this good for? I think the `Triggerer`-trait provides according facility.

Reply:
The Triggerer trait puts signals into the $this->triggered_signals[$event] array which would not allow us to distinguish the two click events. Moving everything to a toggle event would not allow us to link it to two TriggeredSignals in my understanding.

Reply:
Could you just use "toggle_on" and "toggle_off" instead? I mean, the trait doesn't do anything interesting with the signals and actions, just stores them and has them handy for the Renderer.